### PR TITLE
Site Creation : Integrated working Domains suggestion service

### LINKED ISSUE #10669: [Site Creation] Implementation of the Domains Service
Implementation of the DomainsService integrated with the actual backend API.

_Notes:_
- To facilitate parallel development, #10711 introduced a workaround. That workaround should be removed here : see `SiteCreator.swift:42-48`. This will also be noted in #10712.
- There has been some discussions (`p9wMUP-e5-p2`) about properly handling `400` error codes during the integration of this service. We should ensure that's properly addressed here.

### PR DESCRIPTION
**Pull request by stevebaranski**
### Description
Fixes #10669.

To test:

- Checkout the branch and confirm that tests pass.
- This branch was created off of [this PR branch](https://github.com/wordpress-mobile/WordPress-iOS/pull/10784), so only the last 6 commits (of 12) are relevant to this discussion.
- Step through the site creation flow as shown below and verify that the Domains suggestion service responds to your query.
- Alternately toggle network connectivity and validate the error state is shown as expected.

Update release notes:

- [X] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### Visual QA

10669_pr_02

The error states behave the same way as for Verticals. Static screenshots are included below for reference.

10669_iphone_noconnection

10669_iphone_noresults

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Site Creation/Shared/ErrorStates/InlineErrorRetryTableViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+// MARK: - InlineErrorRetryTableViewCellAccessoryView
+
+/// The accessory view comprises a retry Gridicon & advisory text
+///
+private class InlineErrorRetryTableViewCellAccessoryView: UIStackView {
+
+    // MARK: Properties
+
+    /// A collection of parameters uses for view layout
+    private struct Parameters {
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Site Creation/Shared/ErrorStates/InlineErrorRetryTableViewCell.swift]: Nitpick: Same comment as on the verticals PR about perhaps renaming this – `Metrics` or similar?

Comment (stevebaranski) [reply WordPress/Classes/ViewRelated/Site Creation/Shared/ErrorStates/InlineErrorRetryTableViewCell.swift]: I have pushed a commit that addresses this.

**Comment by ctarda** [comment]
After chatting with @frosty it seems like the branch is not building due to a difference of the spelling in a method in `WordPressComServiceRemote`. It is still building on my machine after a clean build, but it crashes on startup now, I assume that, since I have been switching branches quite a bit this morning, some symbols are cached.

**Comment by frosty** [comment]
All changes look good! 👍 The change to the 'no results' message is much clearer, too.

**Comment by stevebaranski** [comment]
Thank you @frosty & @ctarda!

**Comment by frosty** [comment]
Just circling back to this – I realised I forgot to note why I approved it. I'm not entirely sure what the issue was but some combination of Xcode and CocoaPods were failing to pull in the latest version of WordPressKit. Eventually it Just Worked™ after the right combination of restarting, cleaning, and rebuilding.

**Comment by stevebaranski** [comment]
If I were to speculate, I suspect the issue stems from a difference between `rake xcode` and other methods of opening a workspace. Once the companion change to `WordPressKit` was made, the new version of that framework must be retrieved.

**Pull request merged**